SYCL/GroupAlgorithm/SYCL2020/sort.cpp

```diff
@@ -38,14 +36,11 @@ struct CustomFunctor {
   }
 };
 
-// we need it since using std::abs leads to compilation error
-template <typename T> T my_abs(T x) { return x >= 0 ? x : -x; }
```

Comment:
I'm ok with the change if there are no compilation errors related to `abs` as it's mentioned in the comment

Reply:
As you can see in pre-commit logs, there no compilation errors. BTW, the comment is about `std::abs`, not `sycl::abs`.
I trusted your comment and didn't try to use `std::abs` here.

Reply:
I see. Sorry, missed that there is `sycl::abs` now.

Reply:
I checked `std::abs` and I see the problem. 

```c++
llvm-test-suite/SYCL/GroupAlgorithm/SYCL2020/sort.cpp:40:10: error: call to 'abs' is ambiguous
  return std::abs(lhs - rhs) > epsilon;
         ^~~~~~~~
/llvm-test-suite/SYCL/GroupAlgorithm/SYCL2020/sort.cpp:49:9: note: in instantiation of function template specialization 'check<unsigned int>' requested here
    if (check(expected[i], got[i], epsilon)) {
        ^
```

Standard library doesn't provide abs overloads for unsigned integer types. It make sense, so the question is: why SYCL spec defines abs for unsigned integer types?

I suppose your code might not work as expected for unsigned types. I suggest you to think about it.

Reply:
Thanks for pointing it out. What is your opinion whether `sycl::abs` for unsigned integer types is a bug or a feature of SYCL?

Reply:
Technically, I don't any problems with defining such function, but the use case is unclear. This function does nothing, but return its only argument.
I suppose it wasn't intention to define such function. It's just the side effect of listing all supported OpenCL types as allowed parameters for integer functions.
@intel/dpcpp-specification-reviewers, FYI.

Reply:
I captured this question as a Kronos issue against the SYCL spec: https://gitlab.khronos.org/sycl/Specification/-/issues/584